src/parser/ParseException.h

```diff
@@ -46,6 +46,21 @@ class ParseException : public std::exception {
     return metadata_;
   }
 
+  [[maybe_unused]] [[nodiscard]] std::optional<std::string>
```

Comment:
Ok, one of the attributes is wrong as they contradict each other. I think it should only be `[[nodiscard]]`.

Reply:
Ah, but you in fact never use this...
I think you can just remove this info, as everything is in the json/the metadata now.